sdk/core/azure-core/azure/core/credentials.py

```diff
@@ -48,6 +83,26 @@ def get_token(
         ...
 
 
+@runtime_checkable
+class TokenInfoCredential(TokenCredential, Protocol):
+    """Protocol for classes able to provide OAuth access tokens with additional properties."""
+
+    def get_token_info(self, *scopes: str, options: Optional[TokenRequestOptions] = None) -> AccessTokenInfo:
```

Comment:
If I understand correctly, we only need

def get_token_info(self, *scopes: str, options: Optional[dict] = None) -> AccessTokenInfo:

And what is the behavior if we fail to get a token from get_token_info?

Reply:
And to make sure we are on the same page, right after we ship the new protocol, we need another identity GA with the credentials that implement the new protocol.

Reply:
> def get_token_info(self, *scopes: str, options: Optional[dict] = None) -> AccessTokenInfo:

Putting the TypedDict class in the signature will clue people in to what keys and types are expected in the options dict.
Users can still pass in normal dicts with only a subset of the keys and typing will still be happy if the key names and types align.

```cred.get_token_info("scope", options={"tenant_id": 12345})```

> And what is the behavior if we fail to get a token from get_token_info?

If `get_token_info` is defined and fails, it's treated the same as if `get_token` fails. Also, if `get_token_info` were to fail, likely `get_token` would fail as well.

> And to make sure we are on the same page, right after we ship the new protocol, we need another identity GA with the credentials that implement the new protocol.

Yes, I've been working on the identity side of things separately. We can get the identity protocol implementation changes in as soon as possible. However, things will still continue to work even if we GA this and our credentials don't immediately implement the new `get_token_info` method. The `get_token` method will still continue to be used  in the policies.

Reply:
Then how do we prompt the default value of those arguments?

Reply:
`get_token_info` implementations will treat the absence of an `options` parameter to be None (or its falsey value). Pretty much just treated as if it we were extracting the value from a `kwargs` dict.

With the TypeDict class, all fields can be omitted, but if present, it must be of the specified type.

Reply:
I meant we could have get_token(*, enable_cae=False, tenant_id=None, use_default_broker_account=False) in the signature so user could clearly see the default value.

If everything in in options and we only have options=None, how can they know the default value of enable_cae in intellisense?